core/src/consensus.rs

```diff
@@ -174,7 +184,7 @@ pub fn header_version(height: u64) -> HeaderVersion {
 /// Check whether the block version is valid at a given height, implements
 /// 6 months interval scheduled hard forks for the first 2 years.
 pub fn valid_header_version(height: u64, version: HeaderVersion) -> bool {
-	height < 3 * HARD_FORK_INTERVAL && version == header_version(height)
+	height < 4 * HARD_FORK_INTERVAL && version == header_version(height)
 }
```

Comment:
Just realized this logic is only correct for `mainnet`. 
It is not valid for either floonet or `usertesting` which do not use `HARD_FORK_INTERVAL` directly.

This is not specific to this PR and I don't think it has a material impact here - but we should revisit this.

Tracking here - https://github.com/mimblewimble/grin/issues/3335

Reply:
Why can't floonet and usertesting be limited to 2 years worth of block just like mainnet is for version 4? Of course that will be changed downward at the next HF, but for now it looks fine to me.

Reply:
Yeah maybe that is fine actually.
I was originally thinking automatedtesting wants to be able to test an invalid block due to excessive block height. But its going to be based on header version anyway.

:+1: